In [5]:
--ELSO FELADAT
--Jelenítsük meg minden egyes szoba esetén, hogy hányszor foglalták le!

--Csak a szoba azonosítója és a foglalások darabszáma jelenjen meg
--Azok a szobák is jelenjenek meg, ahol nem történt foglalás


SELECT sz.SZOBA_ID,
        IIF(COUNT(*) = 0, 'Nem volt foglalas', COUNT(*)) AS 'Db'
FROM Foglalas f JOIN Szoba sz ON f.SZOBA_FK = sz.SZOBA_ID 
GROUP BY sz.SZOBA_ID

(197 rows affected)

Total execution time: 00:00:00.047

SZOBA_ID Db 1 8 2 4 3 8 4 6 5 4 6 6 7 6 8 5 9 6 10 4 11 4 12 9 13 4 14 3 15 6 16 5 17 6 18 6 19 6 20 5 21 4 22 6 23 3 24 3 25 2 26 2 27 7 28 6 29 6 30 3 31 3 32 1 33 6 34 4 35 9 36 6 37 3 38 8 39 7 40 4 41 2 42 6 43 8 44 6 45 5 46 4 47 2 48 9 49 7 50 4 51 5 52 5 53 2 54 6 55 8 56 2 57 4 58 13 59 7 60 1 61 5 62 5 63 4 64 6 65 1 66 4 67 6 68 3 69 4 70 11 71 3 72 4 73 4 74 9 75 3 76 3 77 6 78 6 79 5 80 7 81 5 82 6 83 8 84 4 85 4 86 5 87 5 88 8 89 8 90 4 91 4 92 6 93 5 94 10 95 4 96 7 97 7 98 4 99 5 100 4 101 3 102 2 103 4 104 4 105 2 106 6 107 5 108 2 109 7 110 8 111 5 112 5 113 5 114 4 115 7 116 2 117 8 118 7 119 5 120 6 121 8 122 9 123 5 124 6 125 1 126 6 127 7 128 5 129 4 130 6 131 5 132 6 133 6 134 6 135 3 136 3 137 6 138 7 139 7 140 8 141 6 142 2 143 7 144 5 145 6 146 6 147 1 148 6 149 7 150 6 151 3 152 4 153 5 154 5 155 5 156 7 157 5 158 3 160 5 161 6 162 3 163 7 164 5 165 7 166 3 167 2 168 5 169 6 170 5 171 5 172 9 173 4 174 4 175 5 176 6 177 4 178 4 179 3 180 3 181 6 182 12 183 4 184 4 185 12 186 4 187 7 188 6 189 5 190 5 191 5 192 2 193 7 194 1 195 6 196 4 197 6 198 8

In [12]:
--MÁSODIK FELADAT
SELECT *
FROM Vendeg
WHERE USERNEV IN
(
    SELECT v.USERNEV
    FROM Vendeg v JOIN Foglalas f ON v.USERNEV=f.UGYFEL_FK
                JOIN Szoba sz ON f.SZOBA_FK = sz.SZOBA_ID
    WHERE sz.FEROHELY = 4 AND sz.KLIMAS = 'i'
    
)

(3 rows affected)

Total execution time: 00:00:00.024

USERNEV NEV EMAIL SZAML_CIM SZUL_DAT JOZSEFG Gyuris József józsef.gyuris@mail.hu 2660 Balassagyarmat Petőfi utca 1/2. 1975-05-26 PETERB Berendi Péter péter.berendi@mail.hu 3980 Sátoraljaújhely Vasút utca 4/10. 1969-01-01 tunde Turcsik Tünde tunde@gmail.com 7130 Tolna Fő út 122. 1974-02-12

HATODIK FELADAT

<span style="color: rgb(181, 182, 192); font-family: Inconsolata; font-weight: 700; white-space: pre-line; background-color: rgb(23, 23, 23);">SET konyvtar 'Corvinus konyvtar'</span>  

<span style="color: rgb(181, 182, 192); font-family: Inconsolata; font-weight: 700; white-space: pre-line; background-color: rgb(23, 23, 23);">HSET konyv:1 cim 'A darts mesterei' iro 'Raymond Van Barneveld' kiadas_eve 2009</span><span style="color: rgb(181, 182, 192); font-family: Inconsolata; font-weight: 700; white-space: pre-line; background-color: rgb(23, 23, 23);"><br></span>

<span class="cli-command-wrapper" data-testid="cli-command-wrapper" style="box-sizing: border-box; margin: 0px; padding: 0px; border: none; vertical-align: baseline; white-space: pre-line; font-family: Inconsolata; color: rgb(181, 182, 192); background-color: rgb(23, 23, 23); font-variant-numeric: normal !important; font-variant-east-asian: normal !important; font-kerning: auto !important; font-optical-sizing: auto !important; font-feature-settings: normal !important; font-variation-settings: normal !important; font-weight: bold !important; font-stretch: normal !important; line-height: 15px !important;">SADD mufaj 'sport' 'horror' 'fantasy'</span><span style="color: rgb(181, 182, 192); font-family: Inconsolata; font-weight: 700; white-space: pre-line; background-color: rgb(23, 23, 23);"><br></span>

<span style="color: rgb(181, 182, 192); font-family: Inconsolata; font-weight: 700; white-space: pre-line; background-color: rgb(23, 23, 23);">RPUSH kikolcsonzott 'A darts mesterei' 'A haz' 'Harry Potter es a bolcsek kove'</span>

<span style="color: rgb(181, 182, 192); font-family: Inconsolata; font-weight: 700; white-space: pre-line; background-color: rgb(23, 23, 23);">ZADD kolcs_szam 3 'A darts mesterei' 2 'A haz' 5 'Harry Potter es a bolcsek kove'</span><span style="color: rgb(181, 182, 192); font-family: Inconsolata; font-weight: 700; white-space: pre-line; background-color: rgb(23, 23, 23);"><br></span>

<span style="color: rgb(181, 182, 192); font-family: Inconsolata; font-weight: 700; white-space: pre-line; background-color: rgb(23, 23, 23);"><br></span>

CREATE TABLE STUDENT

            ... (

            ... ID int,

            ... Nev TEXT,

            ... Teljesitett\_felevek LIST\<INT\>,

            ... Eredmenyek MAP\<INT, DOUBLE\>,

            ... Nyelvtudas SET\<TEXT\>,

            ... PRIMARY KEY(ID)

            ... );

In [ ]:
--HARMADIK FELADAT
--Készítsen lekérdezést, amely sorszámozza a vendégeket a foglalások száma alapján

--A sorszám mellett jelenjen meg a vendég azonosítója, neve és a foglalások száma is
--A legtöbbet foglaló vendég kapja az egyes sorszámot
--Holtverseny esetén a számozás annyit ugorjon, ahány egyforma érték van

SELECT ROW_NUMBER() OVER(ORDER BY ),

FROM Vendeg v JOIN Foglalas f ON v.USERNEV = f.UGYFEL_FK

In [30]:
--NEGYEDIK FELADAT
--Jelenítsük meg, hogy hány foglalás történt szobánként, naponta (METTOL), valamint szobánként, azon belül naponta 

--Szempontonként külön-külön számoljuk a foglalásokat
--A foglalások száma minden esetben legalább 2 legyen

SELECT  sz.SZOBA_ID AS 'Szobánként',
        f.METTOL AS 'Naponként',
        COUNT(*) AS 'Db'
FROM Szoba sz JOIN Foglalas f ON sz.SZOBA_ID = f.SZOBA_FK 
GROUP BY ROLLUP(sz.SZOBA_ID, f.METTOL, (sz.SZOBA_ID, f.METTOL))
HAVING COUNT(*) >= 2

: Msg 144, Level 15, State 1, Line 11
Cannot use an aggregate or a subquery in an expression used for the group by list of a GROUP BY clause.

Total execution time: 00:00:00.002

In [ ]:
--ÖTÖDIK FELADAT
--Készítsünk ideiglenes táblát #MASKED_GUEST néven, ahol 

--a tábla felépítése megegyezik a vendég tábláéval, DE
--a vendégek nevének csak az első 2, és az utolsó 3 karaktere jelenjen meg, közötte X-ek legyenek!

CREATE TABLE #MASKED_GUEST
(
    USERNEV NVARCHAR(20) PRIMARY KEY,
    NEV NVARCHAR(50) MASKED WITH(FUNCTION = 'partial(2, "X", 3)'),
    EMAIL NVARCHAR(60),
    SZAML_CIM NVARCHAR(100),
    SZUL_DAT DATE
)